### PR TITLE
Modify create/update detector page for new UI design

### PR DESCRIPTION
**Pull request by kaituo**
*Issue #, if available:*

*Description of changes:*
create_1
create_2

This PR modifies previous create/update detector page:
First, have instruction text above the form field according to Polaris guidance.
Second, update instruction, placeholder, and button’s content.
Third, reformat text and define the new format in a new scss file.
Fourth, move the data filter as part of the data source.
Fifth, add detector operation settings, including the detector interval and window delay.
Sixth, make detector description optional.
Eighth, update test snapshot and fix outdated tests.

Testing n the done:
*unit tests pass.
*manually verified that I could create and update detector

Note: The newly added unit tests and some other 16 test suites fail after the recent commit adds @babel/transform-runtime. Without that commit, all tests pass.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: public/pages/createDetector/components/createDetector.scss [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+$font-stack: Helvetica Neue, Helvetica, Arial, sans-serif;
```
Comment (mihirsoni) [inline public/pages/createDetector/components/createDetector.scss]: We may not need this.

File: public/pages/createDetector/components/DetectorInfo/DetectorInfo.tsx [hunk 2]
```diff
@@ -25,23 +25,26 @@ interface DetectorInfoProps {
...
-            label=" Detector name"
+            label={
+              <div>
+                <p>Name</p>
+                <p className="sublabel">
+                  Specify a unique and descriptive name that is easy to
+                  recognize.
+                </p>
+              </div>
+            }
```
Comment (mihirsoni) [inline public/pages/createDetector/components/DetectorInfo/DetectorInfo.tsx]: We will need to define some spacing and fonts looks different. Could you verify with Lovinda and show her this screenshot? I believe we need to smaller the text for hints.

Comment (mihirsoni) [reply public/pages/createDetector/components/DetectorInfo/DetectorInfo.tsx]: Another recommendation , as this is the pattern I would suggest to create new Component and use it here. 

A component could take two arguments , title and hint and consume it here.

File: public/pages/createDetector/components/DataFilters/DataFilter.tsx [hunk 3]
```diff
@@ -36,53 +28,46 @@ interface DataFilterProps {
...
+                <p className="sublabel">
+                  Choose a subset of your data source to focus your data stream
+                  and reduce noisy data.
+                </p>
+                <p className="sublabel">
+                  Use the visual editor to create a simple filter, or use the
+                  Elasticsearch query DSL to create more advanced filters.
+                </p>
+              </div>
+            }
```
Comment (mihirsoni) [inline public/pages/createDetector/components/DataFilters/DataFilter.tsx]: Recommend to use Text from EUI to be consistent with sizing and their formattings.

File: public/pages/createDetector/components/createDetector.scss (hunk 1 above)
Comment (kaituo) [reply public/pages/createDetector/components/createDetector.scss]: Why don’t we need this line?

File: public/pages/createDetector/components/DataFilters/DataFilter.tsx (hunk 3 above)
Comment (kaituo) [reply public/pages/createDetector/components/DataFilters/DataFilter.tsx]: I tried to use EUIText, but it is not easy to make it work with EuiFormRow, so I did these formatting inside label.

File: public/pages/createDetector/components/DetectorInfo/DetectorInfo.tsx (hunk 2 above)
Comment (kaituo) [reply public/pages/createDetector/components/DetectorInfo/DetectorInfo.tsx]: The screenshot looks different than what it really does.  I updated screenshots. Also, I created a new component as you said.

File: public/pages/createDetector/containers/CreateDetector.tsx [hunk 4]
```diff
@@ -118,6 +115,8 @@ export function CreateDetector(props: CreateADProps) {
   };
 
   const handleSubmit = async (values: ADFormikValues, formikBag: any) => {
+    console.log('hello');
+    console.log(values);
```
Comment (yizheliu-amazon) [inline public/pages/createDetector/containers/CreateDetector.tsx:119]: nit: can remove those lines for debug use

Comment (kaituo) [reply public/pages/createDetector/containers/CreateDetector.tsx]: removed.  Thanks for the catch.

**Pull request merged**